py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/convert.py

```diff
@@ -4,81 +4,63 @@
-
-    # Enclose entire function in try block, Process can't fail silently.
-    # Must return either pyarrow table or filename
+    logging.info("Converting %s to Parquet Table" % filename)
     try:
-        table = {key.name:[] for key in config.export_schema}
-
         with gzip.open(filename, 'rb') as f:
-            json_data = json.loads(f.read())
-
-            # parse timestamp info out of the header
-            header = json_data['header']
-            feed_timestamp = header['timestamp']
-            timestamp = datetime.utcfromtimestamp(feed_timestamp)
-
-            # for each entity in the list, create a record, add it to the table
-            for entity in json_data['entity']:
-                record = config.record_from_entity(entity=entity)
-                record.update({
-                    'year': timestamp.year,
-                    'month': timestamp.month,
-                    'day': timestamp.day,
-                    'hour': timestamp.hour,
-                    'feed_timestamp': feed_timestamp
-                })
-
-                for key in record:
-                    table[key].append(record[key])
+            json_data = json.load(f)
+            pa_table = _json_to_pyarrow(json_data=json_data, config=config)
 
-        ret_obj = pa.table(table, schema=config.export_schema)
+        return pa_table
 
-    except Exception as e:
-        logging.exception(
-            "Error occured converting %s to Parquet File." % filename.name)
-        ret_obj = filename
+    except Exception as exception:
+        logging.exception("Error converting %s" % filename)
+        return filename
 
-    # Send pyarrow table or filename back to main Process for concatenation
-    return ret_obj
 
-def convert_files(filepaths: list[Path], config: Configuration) -> pa.Table:
-    logging.info("Creating pool with %d threads" % MULTIPROCESSING_POOL_SIZE)
-    pool = Pool(MULTIPROCESSING_POOL_SIZE)
-    workers = pool.starmap_async(gz_to_pyarrow,
-                                 [(f, config) for f in filepaths])
-
-    # Collect pyarrow tables from Processes and concat
-    pa_table = None
-    failed_ingestion = []
-
-    for ret_obj in workers.get():
-        if isinstance(ret_obj, pa.Table):
-            if pa_table is None:
-                pa_table = ret_obj
-            else:
-                pa_table = pa.concat_tables([pa_table,ret_obj])
-        else:
-            failed_ingestion.append(ret_obj)
-
-    logging.info(
-        "Completed converting %d files with config %s" % (len(filepaths),
-                                                          config.config_type))
-
-    if len(failed_ingestion) > 0:
-        logging.warning("Unable to process %d files" % len(failed_ingestion))
-
-    return pa_table
+def s3_to_pyarrow(filename: str, config: Configuration):
+    logging.info("Converting %s to Parquet Table" % filename)
+    try:
+        s3_fs = fs.S3FileSystem()
```

Comment:
`gz_to_pyarrow` and `s3_to_pyarrow` are ideentical except for the filesystem being used. Pyarrow also offers a `LocalFileSystem` interface, we could probably combine these functions and just choose the filesystem interface base on the `filename` string???